Objects/methodobject.c

```diff
@@ -164,9 +164,9 @@ meth_dealloc(PyCFunctionObject *m)
     if (m->m_weakreflist != NULL) {
         PyObject_ClearWeakRefs((PyObject*) m);
     }
+    Py_XDECREF(PyCFunction_GET_CLASS(m));
```

Comment:
Would be nice to add a short comment about `PyCFunction_GET_CLASS()` which should be called before any `Py_XDECREF()` because it accesses `PyMethodDef` which can be stored in m_self or m_module.

And just for the case I would move `Py_VISIT(PyCFunction_GET_CLASS(m))` in `meth_traverse()` too (no comment is needed).

Reply:
Done. Is this clear enough?